src/app/admin/websiteDataTable/WebsiteDataTable.tsx

```diff
@@ -0,0 +1,231 @@
+import { fetchWebsiteData, updateDbWebsiteData } from "./FetchWebsiteData";
+import CustomComponent from "./CustomEditComponent";
+import { logErrorReturnLogId, logInfoReturnLogId } from "@/logger/logger";
+
+export interface WebsiteDataRow {
+    readableName: string;
+    id: string;
+    value: string;
+}
+
+const WebsiteDataTable: React.FC = () => {
+    const [rows, setRows] = useState<WebsiteDataRow[]>([]);
+    const [rowModesModel, setRowModesModel] = useState<GridRowModesModel>({});
+    const [isLoading, setIsLoading] = useState<boolean>(true);
+    const apiRef = useGridApiRef();
+
+    useEffect(() => {
+        fetchWebsiteData()
+            .then((response) => setRows(response))
+            .catch(async (error) => {
+                const logId = await logErrorReturnLogId("Error with fetch: website data", error);
+                throw new DatabaseError("fetch", "website data table", logId);
+            })
+            .finally(() => setIsLoading(false));
+    }, []);
+
+    useEffect(() => {
+        // This requires that the DB table has Realtime turned on
+        const subscriptionChannel = supabase
+            .channel("website-data-table-changes")
+            .on(
+                "postgres_changes",
+                { event: "*", schema: "public", table: "website_data" },
+                async () => {
+                    try {
+                        const websiteData = await fetchWebsiteData();
+                        setRows(websiteData);
+                    } catch (error) {
+                        if (error) {
+                            const logId = await logErrorReturnLogId(
+                                "Error with fetch: website data subscription",
+                                error
+                            );
+                            setRows([]);
+                            throw new DatabaseError("fetch", "website data table", logId);
+                        }
+                    }
+                }
+            )
+            .subscribe(async (status, err) => {
+                if (status === "TIMED_OUT") {
+                    const logId = await logErrorReturnLogId(
+                        "Channel Timed Out: Subscribe to website_data table",
+                        err
+                    );
+                    throw new DatabaseError("fetch", "website data table", logId);
+                } else if (status === "CHANNEL_ERROR") {
+                    const logId = await logErrorReturnLogId(
+                        "Channel Error: Subscribe to website_data table",
+                        err
+                    );
+                    throw new DatabaseError("fetch", "website data table", logId);
+                } else if (status === "CLOSED") {
+                    logInfoReturnLogId("Subscription to website_data table closed");
+                } else {
+                    logInfoReturnLogId("Subscribed to website_data table");
+                }
```

Comment:
is this copied over from somewhere? We should commonise the logic if so.

Reply:
Yeah I copied it from packing slots table. I'll commonise the logic, and update WebsiteDataTable.tsx and PackingSlotsTable.tsx to use it. Should I then put this logic in the other pages where we have realtime subscriptions?

Reply:
@rickaw do you want me to update the other pages with realtime subscriptions?

Reply:
That'd be great if it's not too much trouble!